tests/test_simple_indexer.py

```diff
@@ -0,0 +1,170 @@
+    indexer.fill_embedding(search_docs)
+    assert search_docs['a'].embedding is not None
+
+    with pytest.raises(KeyError, match="'b'"):
+        indexer.fill_embedding(DocumentArray([Document(id='b')]))
+
+
+def test_load(tmpdir, docs):
+    metas = {'workspace': str(tmpdir)}
+    indexer1 = SimpleIndexer(metas=metas)
+    indexer1.index(docs)
+    indexer2 = SimpleIndexer(metas=metas)
+    assert_document_arrays_equal(indexer2._storage, docs)
+
+
+def test_index(tmpdir, docs):
+    metas = {'workspace': str(tmpdir)}
+
+    # test general/normal case
+    indexer = SimpleIndexer(metas=metas)
+    indexer.index(docs)
+    assert_document_arrays_equal(indexer._storage, docs)
+
+    # test index empty docs
+    shutil.rmtree(tmpdir)
+    indexer = SimpleIndexer(metas=metas)
+    indexer.index(DocumentArray())
+    assert not indexer._storage
+
+
+def test_delete(tmpdir, docs):
+    metas = {'workspace': str(tmpdir)}
+
+    # index docs first
+    indexer = SimpleIndexer(metas=metas)
+    indexer.index(docs)
+    assert_document_arrays_equal(indexer._storage, docs)
+
+    # delete empty docs
+    indexer.delete({})
+    assert_document_arrays_equal(indexer._storage, docs)
+
+    # delete first 3 docs
+    parameters = {'ids': [f'doc{i}' for i in range(1, 4)]}
+    indexer.delete(parameters)
+    assert_document_arrays_equal(indexer._storage, docs[3:])
+
+    # delete the rest of the docs stored
+    parameters = {'ids': [f'doc{i}' for i in range(4, 7)]}
+    indexer.delete(parameters)
+    assert not indexer._storage
+
+    # delete from empty storage
+    parameters = {'ids': [f'doc{i}' for i in range(4, 7)]}
+    indexer.delete(parameters)
+    assert not indexer._storage
+
+
+def test_update(tmpdir, docs, update_docs):
+     metas = {'workspace': str(tmpdir)}
+
+     # index docs first
+     indexer = SimpleIndexer(metas=metas)
+     indexer.index(docs)
+     assert_document_arrays_equal(indexer._storage, docs)
+
+     # update first doc
+     indexer.update(update_docs)
+     assert indexer._storage[0].id == 'doc1'
+     assert (indexer._storage[0].embedding == [0, 0, 0, 1]).all()
+
+
+@pytest.mark.parametrize('metric', ['euclidean', 'cosine', 'sqeuclidean', 'hamming'])
+def test_search(tmpdir, metric, docs):
+    metas = {'workspace': str(tmpdir)}
+    match_args = {'metric': metric, 'use_scipy': 'metric' != 'hamming'}
+
+    # test general/normal case
+    indexer = SimpleIndexer(match_args=match_args, metas=metas)
+    indexer.index(docs)
+    search_docs = deepcopy(docs)
+    indexer.search(search_docs)
+    for i in range(len(docs)):
+        assert search_docs[i].matches[0].id == f'doc{i + 1}'
+
+    # test search from empty indexed docs
+    shutil.rmtree(tmpdir)
+    indexer = SimpleIndexer(metas=metas)
+    indexer.index(DocumentArray())
+    indexer.search(docs)
+    for doc in docs:
+        assert not doc.matches
+
+    # test search empty docs
+    indexer.search(DocumentArray())
```

Comment:
remove this test

Reply:
Addressed, please see [here](https://github.com/jina-ai/executor-simpleindexer/pull/1/commits/3aee3434b2a5fae552a35014e6fb32509f9f9c03) for the change.